docs/how-to-create-a-machine.md

```diff
@@ -165,3 +239,129 @@ In practice, a `Topology` is a list of pairs, where every pair contains a vertex
 In fact, we use [`singletons`](https://hackage.haskell.org/package/singletons) to promote `Topology` to the type level and use it to keep track of the available machine transitions at the type level.
 
 We use a `Topology vertex` kind, and we construct types of that kind using the `Topology :: [(vertex, [vertex])] -> Topology vertex` type constructor. In such a use, notice that also `vertex` is a kind.
+
+#### Defining a `Topology`
+
+To define a `Topology`, we first need to define the type of its vertices.
+
+This is usually done by defining an enumeration of the vertices, like
```

Comment:
```suggestion
A straightforward approach is to create a sum type of vertices:
```

Reply:
I also thought initially to say `sum type`, but also `Either Int String` is a sum type and that probably won't work here because then it won't satisfy the `Demote vertex ~ vertex` constraint we have in the `Basic` constructor for `StateMachine`.

In fact, I need to add some more information about those constraints and what they imply (basically they imply that the type must be finite)